### ISSUE TITLE
Implement TF-IDF similarity computation

### ISSUE THREAD

**Issue by aaronspindler** [aaronspindler/aaronspindler.com#201]
## Parent Issue
Part of #199 - Related Posts Recommendation Engine

## Description
Implement the core ML logic to compute content similarity between blog posts using TF-IDF and cosine similarity.

## Tasks
- [ ] Install `scikit-learn` (add to requirements.txt)
- [ ] Create `blog/recommendations.py` module
- [ ] Implement `ContentSimilarityEngine` class:
  ```python
  from sklearn.feature_extraction.text import TfidfVectorizer
  from sklearn.metrics.pairwise import cosine_similarity
  
  class ContentSimilarityEngine:
      def __init__(self):
          self.vectorizer = TfidfVectorizer(
              stop_words='english',
              max_features=1000,
              ngram_range=(1, 2)
          )
      
      def compute_similarity_matrix(self, posts):
          # Extract text from posts
          # Vectorize using TF-IDF
          # Compute cosine similarity
          pass
      
      def get_top_similar(self, post_idx, n=4):
          # Return top N similar posts
          pass
  ```
- [ ] Preprocess blog post content:
  - Load all blog posts using `get_all_blog_posts()`
  - Extract title + description + content
  - Remove HTML tags
  - Lowercase and normalize
- [ ] Compute similarity matrix (NxN matrix)
- [ ] For each post, extract top 5 most similar
- [ ] Filter out self-matches (similarity = 1.0)

## Technical Notes
- TF-IDF weights: term frequency × inverse document frequency
- Cosine similarity: dot product of normalized vectors
- Consider stemming/lemmatization (optional)
- Matrix can be large; consider sparse matrix storage

## Acceptance Criteria
- [ ] Can compute similarity for all blog posts
- [ ] Returns sensible recommendations (manually verify)
- [ ] Handles edge cases (single post, no similar posts)
- [ ] Unit tests for similarity computation

## Estimated Effort
Medium (~3-4 hours)

## Depends On
- Need to add scikit-learn to requirements.txt

**Comment by aaronspindler** [comment]
**Part of Epic:** #199
